tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotSerializer.cs

```diff
@@ -233,7 +229,7 @@ private static void WriteFieldsInternal(object source, JsonWriter jsonWriter, Ca
                     continue;
                 }
 
-                if (index == 0 && writeFieldsJsonElement)
```

Comment:
This actually breaks functionality in some cases, see snapshot failures in integration tests.

Reply:
I had to re-approve the tests that contain instance fields and I already did that.

Reply:
I had to re-approve the tests that contain instance fields. Despite that, `AsyncInstanceMethod` used to contain both method and line probes and now it only has line probe, so the method probe snapshot is omitted.

Reply:
Sorry, my bad, I didn't really explain myself clearly in this comment. I explained the issue in more detail here: https://github.com/DataDog/dd-trace-dotnet/pull/3406/files#r1009125686

Reply:
Replied there https://github.com/DataDog/dd-trace-dotnet/pull/3406/files#r1009137144. Resolving this one due to redundancy.